src/pke/include/cryptocontext.h

```diff
@@ -98,10 +98,23 @@ class CryptoContextImpl : public Serializable {
     virtual Plaintext MakeCKKSPackedPlaintextInternal(const std::vector<std::complex<double>>& value,
                                                       size_t noiseScaleDeg, uint32_t level,
                                                       const std::shared_ptr<ParmType> params, usint slots) const {
-        Plaintext p;
         const auto cryptoParams = std::dynamic_pointer_cast<CryptoParametersRNS>(GetCryptoParameters());
-        double scFact;
+        if (level > 0) {
+            // validation of level: We need to compare it to multiplicativeDepth, but multiplicativeDepth is not
+            // readily available. so we get is numModuli and use it for calculations
```

Comment:
Actually, what we want to compare it to is numModuli, and it should be if `(level >= numModuli)`. We use multiplicativeDepth so the user can understand the error message more easily. In the case of `FLEXIBLEAUTOEXT`, the check still needs to be `if (level >= numModuli)`, which translates to `if (level > multiplicativeDepth + 1)`

Reply:
@yspolyakov Wanted to double check the following:
In both cases, we want to throw an exception if `level >= numModuli`. In FLEXIBLEAUTOEXT, we have `multiplicativeDepth = numModuli - 2` and otherwise we have `multiplicativeDepth = numModuli - 1`. So the error message in the case of FLEXIBLEAUTOEXT should be `The level value should be less than or equal to multiplicativeDepth + 1`, and otherwise the error message is  `The level value should be less than or equal to multiplicativeDepth`.

Reply:
I agree with @sarojaduality. We should thrown an exception when `level >= numModuli`. My initial recommendation was incorrect.